flocker/control/schema/endpoints.yml

```diff
@@ -124,12 +124,14 @@ definitions:
         '$ref': 'types.json#/definitions/metadata'
       maximum_size:
         '$ref': 'types.json#/definitions/maximum_size'
-    required:
-      # Temporarily required until volume backends settle down and we know
-      # more about what it means to not have a primary manifestation.
-      - primary
     additionalProperties: false
 
+  configuration_dataset_primary:
+    description: "The configuration for a new dataset with a primary address"
+    allOf:
+      - "$ref": "#/definitions/configuration_dataset"
```

Comment:
`configuration_dataset` is starting to look more like a "type" than an "endpoint" (if that means anything).  Maybe it should move to the other schema file.  And I think the funny word order, "configuration_dataset" instead of "dataset_configuration" was meant as a reflection of which endpoint this related to - the `/configuration/dataset` endpoint.  So perhaps the schema in the type file should be "dataset_configuration" and the schema that remains here, `configuration_dataset_primary`, should just be called `configuration_dataset` - since it is, after all, the schema that `/configuration/dataset` now uses.